Prepare dashboards, panels, and visualizations for Grafana Cloud mount

jdbaldry opened the pull request:
1. Replaces all usage of `relref` shortcodes with `docs/reference` shortcode links instead.
1. Also replaces all usage of `relref` shortcodes containing only an anchor with the raw Markdown equivalent. The `relref` shortcode doesn't check anchor existence and is just extra characters.

jdbaldry committed:
Update dashboards pages to use `docs/reference` shortcode

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

jdbaldry committed:
Update panels-visualizations pages to use `docs/reference` shortcode

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

jdbaldry commented on docs/sources/dashboards/assess-dashboard-usage/index.md:
```diff
@@ -25,7 +25,7 @@ weight: 200
 
 {{% admonition type="note" %}}
-Available in [Grafana Enterprise]({{< relref "../../introduction/grafana-enterprise/" >}}) and [Grafana Cloud Pro and Advanced](/docs/grafana-cloud/). Grafana Cloud insights logs include additional fields with their own dashboards. Read more in the [Grafana Cloud documentation](/docs/grafana-cloud/usage-insights/).
+Available in [Grafana Enterprise][] and [Grafana Cloud Pro and Advanced](/docs/grafana-cloud/). Grafana Cloud insights logs include additional fields with their own dashboards. Read more in the [Grafana Cloud documentation](/docs/grafana-cloud/usage-insights/).
```
I'm gonna have to change this one because reference style links don't work inside admonition shortcodes.

How important is the link to [Grafana Enterprise](https://grafana.com/docs/grafana/latest/introduction/grafana-enterprise/) in this note?

imatwawana replied:
Is it possible to use `/docs/grafana/introduction/grafana-enterprise/` for this? Because this is pretty standard text throughout docs, I'm hesitant to remove the Enterprise links entirely. But if there isn't a clean way to do it, then I think we take it out in these spots.

jdbaldry replied:
Yeah, I think I'm gonna have to see if I can find a way. The need to link from admonitions is too strong for it to be something we can't do.

jdbaldry commented on docs/sources/dashboards/dashboard-public/index.md:
```diff
@@ -150,14 +150,14 @@ If a Grafana user has read access to the parent dashboard, they can view the pub
 
 {{% admonition type="note" %}}
-Available in [Grafana Enterprise]({{< relref "../../introduction/grafana-enterprise/" >}}) and [Grafana Cloud Pro and Advanced](/docs/grafana-cloud).
+Available in [Grafana Enterprise][] and [Grafana Cloud Pro and Advanced](/docs/grafana-cloud).
```
Same here.

jdbaldry commented on docs/sources/dashboards/share-dashboards-panels/index.md:
```diff
@@ -94,14 +94,14 @@ If you created a snapshot by mistake, click **Delete snapshot** to remove the sn
 
 {{% admonition type="note" %}}
-Available in [Grafana Enterprise]({{< relref "../../introduction/grafana-enterprise/" >}}) and [Grafana Cloud](/docs/grafana-cloud/).
+Available in [Grafana Enterprise][] and [Grafana Cloud](/docs/grafana-cloud/).
```
Same here.

jdbaldry commented on docs/sources/dashboards/variables/add-template-variables/index.md:
```diff
@@ -87,32 +87,32 @@ Query variables are generally only supported for strings. If your query returns
 
 {{% admonition type="note" %}}
-Query expressions are different for each data source. For more information, refer to the documentation for your [data source]({{< relref "../../../datasources/" >}}).
+Query expressions are different for each data source. For more information, refer to the documentation for your [data source][].
```
Different link, similar question, how important is the link to [Datasources](https://grafana.com/docs/grafana/latest/datasources/)?

jdbaldry commented on docs/sources/panels-visualizations/query-transform-data/transform-data/index.md:
```diff
@@ -967,10 +967,10 @@ Here is the result after adding a Limit transformation with a value of '3':
 
 {{% admonition type="note" %}}
-This transformation is available in Grafana 9.5+ as an opt-in beta feature. Modify Grafana [configuration file]({{< relref "../../../setup-grafana/configure-grafana/#configuration-file-location" >}}) to enable the `timeSeriesTable` [feature toggle]({{< relref "../../../setup-grafana/configure-grafana/#feature_toggles" >}}) to use it.
+This transformation is available in Grafana 9.5+ as an opt-in beta feature. Modify Grafana [configuration file][] to enable the `timeSeriesTable` [feature toggle][] to use it.
```
Hmm, there's a few of these. I'm gonna have to spend some time thinking about how we can make reference style links work in shortcodes before we can merge this PR.

jdbaldry commented on docs/sources/panels-visualizations/visualizations/table/index.md:
```diff
@@ -150,10 +150,10 @@ If you have a field value that is an image URL or a base64 encoded image you can
 
 {{% admonition type="note" %}}
-This cell type is available in Grafana 9.5+ as an opt-in beta feature. Modify Grafana [configuration file]({{< relref "../../../setup-grafana/configure-grafana/#configuration-file-location" >}}) to enable the `timeSeriesTable` [feature toggle]({{< relref "../../../setup-grafana/configure-grafana/#feature_toggles" >}}) to use it.
+This cell type is available in Grafana 9.5+ as an opt-in beta feature. Modify Grafana [configuration file][] to enable the `timeSeriesTable` [feature toggle][] to use it.
```
Another place

imatwawana reviewed (commented):
I did a spot check and found what looks like some missing updates. Should I be doing more than a spot check on this PR and actually going through each page one-by-one? Let me know.

Sidebar: the way we explained how to set up a docs/reference link isn't the way you're doing it here, so maybe we need to update the writer's toolkit?

jdbaldry committed:
Merge remote-tracking branch 'origin/main' into jdb/2023-08-prepare-panels-visualizations-for-Grafana-Cloud-mount

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

jdbaldry committed:
Use raw Markdown admonitions for contextual links

See https://discourse.gohugo.io/t/markdown-reference-links-in-shortcodes/5770/3.
Should be resolved by https://docs.google.com/document/d/19xd4CD3IrAqQqNR3xQeLfuIV-u2iNSQSBEvPZE9J4qU/edit#heading=h.5sybau7waq2q.

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

jdbaldry committed:
Prefer "Warning" over "Caution"

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

imatwawana commented on docs/sources/dashboards/build-dashboards/_index.md:
```diff
@@ -24,4 +24,9 @@ This section includes the following topics:
 ## Dynamic dashboards
 
-You can create more interactive and dynamic dashboards by adding and using [variables]({{< relref "../variables" >}}). Instead of hard-coding things like server, application, and sensor names in your metric queries, you can use variables in their place. Read more about variables [here]({{< relref "../variables" >}}).
+You can create more interactive and dynamic dashboards by adding and using [variables][]. Instead of hard-coding things like server, application, and sensor names in your metric queries, you can use variables in their place. Read more about variables [here][].
```
This link doesn't work. Rendering as just: [here][]
```suggestion
You can create more interactive and dynamic dashboards by adding and using [variables][]. Instead of hard-coding things like server, application, and sensor names in your metric queries, you can use variables in their place. Read more about variables [here][].
```

imatwawana commented on docs/sources/dashboards/build-dashboards/create-dashboard-url-variables/index.md:
```diff
@@ -20,7 +20,7 @@ weight: 250
 
 Grafana can apply variable values passed as query parameters in dashboard URLs.
-For more information, refer to [Manage dashboard links]({{< relref "../manage-dashboard-links/" >}}) and [Templates and variables]({{< relref "../../variables/" >}}).
+For more information, refer to [Manage dashboard links][] and [Templates and variables][].
```
[Templates and variables] reference is not rendering as a link in the build, but just as the raw text. 
```suggestion
For more information, refer to [Manage dashboard links][] and [Templates and variables][].
```

jdbaldry replied:
Good catch, fixed in 62139350fc

jdbaldry committed:
Prettier

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

jdbaldry committed:
Fix some links missing destinations

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

imatwawana reviewed (commented):
I've checked every link under the Dashboards content and noted the ones that don't link properly or don't link at all. I still have to review the Panels & Visualizations section, which I'll do later today.

jdbaldry committed:
Fix 'time range controls' link

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

jdbaldry committed:
Add missing 'HTTP APIs' link

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

jdbaldry committed:
Fix dashboard links missing destinations

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

jdbaldry commented:
I've pushed a few more commits that fix links missing variables. It looks the the automated script didn't add entries for a second link appearing in the same line.

I don't see the same behavior with regards to the links inside notes. Have you pulled all the latest commits for this branch? I know they didn't work inside the admonitions but they should work inside the raw Markdown notes.

jdbaldry committed:
Fix links missing destinations in panels and visualizations documentation

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

imatwawana reviewed (approved):
Just one dead link but otherwise, I think this is good to go! Thanks for all your work!

jdbaldry committed:
Fix dud link

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>
